src/main/java/bio/terra/app/controller/AdminApiController.java

```diff
@@ -61,4 +77,51 @@ public ResponseEntity<JobModel> registerDrsAliases(List<DrsAliasModel> aliases)
     String jobId = drsService.registerDrsAliases(aliases, userReq);
     return ControllerUtils.jobToResponse(jobService.retrieveJob(jobId, userReq));
   }
+
+  @Override
+  public ResponseEntity<DatasetModel> adminRetrieveDataset(UUID id) {
+    List<DatasetRequestAccessIncludeModel> include =
+        List.of(
+            DatasetRequestAccessIncludeModel.ACCESS_INFORMATION,
+            DatasetRequestAccessIncludeModel.DATA_PROJECT,
+            DatasetRequestAccessIncludeModel.PROFILE,
+            DatasetRequestAccessIncludeModel.PROPERTIES,
+            DatasetRequestAccessIncludeModel.SCHEMA,
+            DatasetRequestAccessIncludeModel.STORAGE);
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    logger.info(
+        String.format(
+            "Verifying resource type admin authorization: %s for resource type: %s and resource id: %s",
+            userReq.getEmail(), IamResourceType.DATASET, id));
+    iamService.verifyResourceTypeAdminAuthorized(
+        userReq, IamResourceType.DATASET, IamAction.ADMIN_READ_SUMMARY_INFORMATION);
+    logger.info(String.format("Retrieving dataset id: %s", id));
+    DatasetModel datasetModel = datasetService.retrieveDatasetModel(id, userReq, include);
+    return ResponseEntity.ok(datasetModel);
+  }
+
+  @Override
+  public ResponseEntity<SnapshotModel> adminRetrieveSnapshot(UUID id) {
+    List<SnapshotRetrieveIncludeModel> include =
+        List.of(
+            SnapshotRetrieveIncludeModel.SOURCES,
+            SnapshotRetrieveIncludeModel.TABLES,
+            SnapshotRetrieveIncludeModel.RELATIONSHIPS,
+            SnapshotRetrieveIncludeModel.ACCESS_INFORMATION,
+            SnapshotRetrieveIncludeModel.PROFILE,
+            SnapshotRetrieveIncludeModel.PROPERTIES,
+            SnapshotRetrieveIncludeModel.DATA_PROJECT,
+            SnapshotRetrieveIncludeModel.CREATION_INFORMATION,
+            SnapshotRetrieveIncludeModel.DUOS);
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    logger.info(
```

Comment:
There's no harm in these logs, but is this logging necessary? Is it a temporary log until the feature has been tested out? Maybe using `debug()` level logging would make more sense, as I don't think we'd normally want to see this log message.

Reply:
This makes sense to me, although I would note that it may make it harder to debug things after the fact since we set our [log levels in prod](https://github.com/broadinstitute/terra-helmfile/blob/2d7f388beaaacd228720b276b5bcdcbac8f4165f/values/app/datarepo/live/prod.yaml#L196) to `info`.